diff --git a/src/platformsupport/input/evdevtouch/qevdevtouchhandler.cpp b/src/platformsupport/input/evdevtouch/qevdevtouchhandler.cpp
index f3cc160b..66b226a8 100644
--- a/src/platformsupport/input/evdevtouch/qevdevtouchhandler.cpp
+++ b/src/platformsupport/input/evdevtouch/qevdevtouchhandler.cpp
@@ -321,6 +321,11 @@ QEvdevTouchScreenHandler::QEvdevTouchScreenHandler(const QString &device, const
         }
         qCDebug(qLcEvdevTouch, "evdevtouch: found ti-tsc, overriding: min X: %d max X: %d min Y: %d max Y: %d",
                 d->hw_range_x_min, d->hw_range_x_max, d->hw_range_y_min, d->hw_range_y_max);
+    } else if(d->hw_name == QLatin1String("generic ft5x06 (79)")) {
+        d->hw_range_x_max = 1024;
+        d->hw_range_y_max = 600;
+        qCDebug(qLcEvdevTouch, "evdevtouch: found ft5x06, overriding: min X: %d max X: %d min Y: %d max Y: %d",
+                d->hw_range_x_min, d->hw_range_x_max, d->hw_range_y_min, d->hw_range_y_max);
     }
 
     bool grabSuccess = !ioctl(m_fd, EVIOCGRAB, (void *) 1);
@@ -761,6 +766,7 @@ void QEvdevTouchScreenData::reportPoints()
         // Generate a screen position that is always inside the active window
         // or the primary screen.  Even though we report this as a QRectF, internally
         // Qt uses QRect/QPoint so we need to bound the size to winRect.size() - QSize(1, 1)
+
         const qreal wx = winRect.left() + tp.normalPosition.x() * (winRect.width() - 1);
         const qreal wy = winRect.top() + tp.normalPosition.y() * (winRect.height() - 1);
         const qreal sizeRatio = (winRect.width() + winRect.height()) / qreal(hw_w + hw_h);
